serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc

```diff
@@ -1 +1,700 @@
+      flows:
+        clientCredentials: <9>
+          authorizationUrl: http://localhost:8281/auth/realms/kogito/protocol/openid-connect/auth
+          tokenUrl: http://localhost:8281/auth/realms/kogito/protocol/openid-connect/token
+          scopes: { }
+----
+
+<1> REST path to access the `exchangeRate` operation in the remote server.
+<2> `exchangeRate` operation parameters.
+<3> `exchangeRate` operation responses.
+<4> Response type and data exchange format.
+<5> Specifies that the `exchangeRate` operation is secured with the `acme-financial-oauth` security scheme.
+<6> Response type specification.
+<7> `acme-financial-oauth` security scheme specification.
+<8> Security scheme type. This information indicates that you must configure a {quarkus_guide_security_openid_connect_client_url}[Quarkus OpenId Connect Client (OIDC)] with name `acme_financial_oauth` to execute the operation.
+<9> Authentication flow to use, and flow information.
+--
+
+[NOTE]
+====
+You can find the full implementation of this service in the `serverless-workflow-oauth2-orchestration-quarkus/acme-financial-service` directory.
+====
+
+== Configuration
+
+Most relevant configurations for the Serverless Workflow:
+
+.Example application.properties file
+[source, properties]
+----
+quarkus.openapi-generator.codegen.spec.acme_financial_service_yml.base-package=com.acme <1>
+
+quarkus.rest-client.acme_financial_service_yml.url=http://localhost:8483 <2>
+
+quarkus.oidc-client.acme_financial_oauth.discovery-enabled=false <3>
+quarkus.oidc-client.acme_financial_oauth.auth-server-url=http://localhost:8281/auth/realms/kogito/protocol/openid-connect/auth <4>
+quarkus.oidc-client.acme_financial_oauth.token-path=http://localhost:8281/auth/realms/kogito/protocol/openid-connect/token <5>
+quarkus.oidc-client.acme_financial_oauth.client-id=kogito-app <6>
+quarkus.oidc-client.acme_financial_oauth.grant.type=client
+quarkus.oidc-client.acme_financial_oauth.credentials.client-secret.method=basic <7>
+quarkus.oidc-client.acme_financial_oauth.credentials.client-secret.value=secret <8>
+----
+
+<1> Is the package name for the automatically generated classes that implements the access to the all operations defined in the file `acme-financial-service.yml`.
+<2> Is the root URL to access all the operations defined in the file `acme-financial-service.yml`. For the `exchangeRate` operation, an absolute URL such as \http://localhost:8483/financial-service/exchange-rate is automatically generated.
+
+<3> Disable the Oauth2 server endpoints discovery since the endpoints provided in the <<acme-financial-service-yml, acme-financial-service.yml>> file will be used instead.
+<4> Is the authentication URL of the Oauth2 server.
+<5> Is the relative path or absolute URL of the Oauth2 token endpoint which issues access and refresh tokens.
+<6> Is the client-id to identify the Serverless Workflow against the authorization service, such as `kogito-app`. This identifier must be provided by Acme.
```

Comment:
At this point of the guide, the reader already knows that the example is about to use the "Acme Financial Services" that are provided by Acme.

Acme is normally a fictitious company name, that is many times used around different examples in the computer world, and other areas. The name comes from an old, but very funny cartoon.

In this context, I mean that the identifier to authenticate the SW against Acme's Oauth2 server, is provided by Acme. This is because Acme is the owner that server.

Reply:
I know, but even reading the full document I had this question because, for me, it was not clear what By Acme really means, wouldn't be better stick with something similar to Quarkus guide? is the client id used to validate the token?

Reply:
ok.